oceannavigator/frontend/src/components/TimePicker.jsx

```diff
@@ -117,142 +109,134 @@ class TimePicker extends React.Component {
-                        _("Sep."),
-                        _("Oct."),
-                        _("Nov."),
-                        _("Dec.")
-                      ]
-                    }
-                  });
-                  break;
-                case "hour":
-                  $(this.refs.picker).datepicker({
-                    Button: false,
-                    dateFormat: "dd MM yy",
-                    onClose: this.pickerChange,
-                    minDate: new Date(map[min]),
-                    maxDate: new Date(map[max])
-                  });
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "minDate",
-                    new Date(map[min])
-                  );
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "maxDate",
-                    new Date(map[max])
-                  );
-                  break;
-                case "day":
-                default:
-                  $(this.refs.picker).datepicker({
-                    Button: false,
-                    dateFormat: "dd MM yy",
-                    onClose: this.pickerChange,
-                    minDate: new Date(map[min]),
-                    maxDate: new Date(map[max])
-                  });
-
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "minDate",
-                    new Date(map[min])
-                  );
-                  $(this.refs.picker).datepicker(
-                    "option",
-                    "maxDate",
-                    new Date(map[max])
-                  );
-                  break;
-              }
-              this.pickerChange();
-            }.bind(this)
+    // eslint-disable-next-line max-len
+    GetTimestampsPromise(props.dataset, props.variable).then(timestampResult => {
+      const data = timestampResult.data;
+
+      let map = {};
+      let revmap = {};
+
+      for (let i = 0; i < data.length; ++i) {
+        const d1 = new Date(data[i].value);
+        const d2 = new Date(d1.getTime() + d1.getTimezoneOffset() * 60000);
+        let d3 = d2;
+        if (this.props.quantum !== "hour") {
+          d3 = new Date(
+            Date.UTC(
+              d1.getUTCFullYear(),
+              d1.getUTCMonth(),
+              d1.getUTCDate(),
+              0,
+              0,
+              0,
+              0
+            )
           );
-        }.bind(this),
-
-        error: function (xhr, status, err) {
-          console.warn(props.url, status, err.toString());
-        }.bind(this)
-      });
+        }
+        map[data[i].id] = d2;
+        revmap[d3.toUTCString()] = data[i].id;
+      }
 
-    }
+      const [min, max] = this.getMinMaxTimestamps(props, data);
+
+      this.setState(
+        {
+          data: data,
+          map: map,
+          revmap: revmap,
+          min: min,
+          max: max
+        },
+        () => {
```

Comment:
Switch to using arrow function so we don't need to explicitly call .bind(this) since arrow functions are automatically bound to the parent object